protobuf-maven-plugin/src/main/java/io/github/ascopes/protobufmavenplugin/sources/incremental/SerializedIncrementalCacheModule.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2023 - 2024, Ashley Scopes.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.ascopes.protobufmavenplugin.sources.incremental;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.KeyDeserializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Path;
+
+/**
+ * Custom serializer/deserializer support for the incremental cache serialization in Jackson.
+ *
+ * @author Ashley Scopes
+ * @since 2.7.0
+ */
+final class SerializedIncrementalCacheModule extends SimpleModule {
```

Comment:
- [ ] Check if a simpler solution is possible by restructuring the data being stored, or by storing URI objects/String objects instead of Path objects as keys directly (since we will always be on the root file system anyway).